Common/TableProducer/occupancyTableProducer.cxx

```diff
@@ -0,0 +1,1053 @@
+///         Ambg tracks were not used
+/// \author Rahul Verma (rahul.verma@iitb.ac.in) :: Marian I Ivanov (marian.ivanov@cern.ch)
+
+#include <vector>
+#include <unordered_map>
+#include <algorithm>
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/O2DatabasePDGPlugin.h"
+
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/Centrality.h"
+#include "Common/DataModel/Multiplicity.h"
+
+#include "PWGLF/DataModel/LFStrangenessTables.h"
+
+#include "Common/DataModel/OccupancyTables.h"
+
+#include "Framework/AnalysisDataModel.h"
+#include "Framework/HistogramRegistry.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "DataFormatsFT0/Digit.h"
+#include "DataFormatsParameters/GRPLHCIFData.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+int32_t nBCsPerOrbit = o2::constants::lhc::LHCMaxBunches;
+const int nBCinTF_grp80 = 1425;
+
+struct occTableProducer {
+
+  Service<o2::ccdb::BasicCCDBManager> ccdb;
+
+  // declare production of tables
+  Produces<aod::OccIndexTable> GenOccIndexTable;
+  Produces<aod::Occs> GenOccTable;
+
+  Configurable<int> customOrbitOffset{"customOrbitOffset", 0, "customOrbitOffset for MC"};
+
+  // Histogram registry;
+  HistogramRegistry recoEvent{"recoEvent", {}, OutputObjHandlingPolicy::AnalysisObject, true, true};
+
+  void init(InitContext const&)
+  {
+    // Getting Info from CCDB, to be implemented Later
+    recoEvent.add("h_nBCinTF", "h_nBCinTF(to check nBCinTF)", {HistType::kTH1F, {{100, 114040, 114060}}}); // 114048
+    recoEvent.add("h_bcInTF", "h_bcInTF", {HistType::kTH1F, {{2000, 0, 200000}}});
+    recoEvent.add("h_RO_T0V0Prim_Unfm_80", "h_RO_T0V0Prim_Unfm_80:median contributors", {HistType::kTH1F, {{12 * 2, -1, 11}}});
+    recoEvent.add("h_RO_FDDT0V0Prim_Unfm_80", "h_RO_FDDT0V0Prim_Unfm_80:median contributors", {HistType::kTH1F, {{12 * 2, -1, 11}}});
+    recoEvent.add("h_RO_NtrackDet_Unfm_80", "h_RO_NtrackDetITS/TPC/TRD/TOF_80:median contributors", {HistType::kTH1F, {{12 * 2, -1, 11}}});
+
+    recoEvent.print();
+  }
+
+  template <typename T>
+  bool vectorAscendingSortCheck(const T& vec)
+  {
+    for (uint i = 1; i < vec.size(); i++) {
+      if (vec[i] < vec[i - 1]) {
+        LOG(debug) << "DEBUG :: Vector unsorted at Position = " << i;
+        return false;
+      }
+    }
+    return true;
+  }
+
+  void GetNormalisedVector(const std::vector<float>& OriginalVec, std::vector<float>& NormVec, const float& scaleFactor)
+  {
+    NormVec.resize(OriginalVec.size());
+    std::transform(OriginalVec.begin(), OriginalVec.end(), NormVec.begin(), [scaleFactor](float x) { return x * scaleFactor; });
+  }
+
+  void GetMedianOccVect(std::vector<float>& medianVector, std::vector<std::vector<int>>& medianPosVec, const std::vector<std::vector<float>>& vectors)
```

Comment:
Modified this structure, now vector references are passed using variadic template so that no vectors are copied.
The vectors which are passed, all have fixed size